plugins/discovery-azure-classic/src/test/java/org/elasticsearch/discovery/azure/classic/AzureDiscoveryClusterFormationTests.java

```diff
@@ -99,6 +99,9 @@ public static void setupKeyStore() throws IOException {
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
+        if (inFipsJvm()) {
```

Comment:
The issue with this one is that it has many `@BeforeClass` already and there is no guarantee the `assumeFalse` will be ran before those. `startHttp` in particular  will throw an error in a FIPS JVM, and so will the `stopHttpd()` that will run on cleanup because of `@AfterClass`. I discussed possibilities to mute this in a cleaner way with Tal and he suggested trying with `ESIntegTestCase.ClusterScope(scope = Scope.TEST, numDataNodes = 0)` so that no nodes are setup so I can try this if the current solution is too ugly.

Reply:
Can you do the one before the class in a ClassRule? We use it in a few places for cases like this. One recent case that was a bit different is https://github.com/elastic/elasticsearch/pull/30763 and I believe there is another case in SecurityIntegTestCase

Reply:
Thanks, I wasn't aware of this